docs/pages/core-concepts.mdx

```diff
@@ -46,61 +47,87 @@ Services**, which can run in private networks. This means that, in the Proxy
+applications or the AWS Console. Protect your internal tools and dashboards.
 
 Read more about the [Teleport Application
 Service](./enroll-resources/application-access/introduction.mdx).
 
 ### Teleport Database Service
 
 Proxies TCP traffic in the native protocols of popular databases, including
-PostgreSQL and MySQL.
+PostgreSQL and MySQL. Use to enforce role-based access to databases with full audit logging.
 
 Read more about the [Teleport Database
 Service](./enroll-resources/database-access/database-access.mdx).
 
+### Teleport Discovery Service
+
+The Teleport Discovery Service automates the process of finding and enrolling cloud resources such as AWS EC2 instances and Azure VMs into a 
+Teleport cluster. It continuously scans supported cloud environments and adds matching resources based on configured rules, 
+reducing the need for manual node registration. 
+
+Read more about the [Teleport Discovery Service](./enroll-resources/auto-discovery/auto-discovery.mdx).
+
 ### Teleport Desktop Service
 
-Proxies Remote Desktop Protocol traffic to Windows desktops.
+Proxies Remote Desktop Protocol (RDP) traffic to Windows desktops. Enables access control and session recording for Windows systems
 
 Read more about the [Teleport Desktop
 Service](./enroll-resources/desktop-access/introduction.mdx).
 
-### Teleport Kubernetes Service
+### Teleport Jamf Service
 
-Proxies HTTP traffic to the Kubernetes API server.
+The Jamf Service integrates with Jamf to enrolls trusted devices for macOS endpoints. 
+Before a user is granted access, Teleport checks the device's compliance status with Jamf. 
+This service is ideal for organizations that need to verify macOS device posture as part of 
+their access control strategy, ensuring only trusted devices are allowed to connect. 
+[Read more about the Jamf Service](./identity-governance/device-trust/jamf-integration.mdx).
 
-Read more about the [Teleport Kubernetes
-Service](./enroll-resources/kubernetes-access/introduction.mdx)
+### Teleport Debug Service
 
-### Teleport SSH Service
+Provides internal diagnostics, exposing metrics, performance profiling tools, and debugging endpoints. 
+It is designed for measuring performance, troubleshooting scenarios, and enabling operators to change the logging level without restarting teleport. 
+This can be helpful for analyzing bottlenecks, or diagnosing unexpected behavior in a controlled environment. The debug service is exposed locally by default.
+[Read more about the Debug Service](./admin-guides/management/admin/troubleshooting.mdx#step-13-enable-verbose-logging).
 
-An SSH server implementation that allows users to execute commands on remote
-machines while taking advantage of Teleport's built-in access controls,
-auditing, and session recording.
+## Agent
 
-Read more about the [Teleport SSH Service](./enroll-resources/server-access/introduction.mdx).
+A Teleport instance that runs one or more services to provide access to multiple resources. 
```

Comment:
```suggestion
A Teleport instance that runs one or more services to provide access to infrastructure resources. 
```
Since an agent could theoretically protect a single resource. Also clarifying "infrastructure" versus other kinds of resource